### PR TITLE
Upload/save to root path

### PR DESCRIPTION
**Pull request by jotakar**
Now you can not upload/saves files to root path using fs service. Server sent an error 400, due to an empty path
With this modif we add a ./ if the path is empty.

### CODE REVIEW

**Comment by jotakar** [comment]
Could anybody check and commit this request?
Thanks

**Comment by lexoyo** [comment]
Looks good to me!
@jbIPS ?
@clemos ?

**Comment by clemos** [comment]
since it's kind of hacky (changing req.params, which IMHO should be considered readonly / immutable...), 
I think you should at least add a comment to document what is going on here ;)

**Comment by clemos** [comment]
or something like : 
```javascript
const target = { 
    filename: req.params[0], // or whatever req.params[0] is...
    folder = req.params[1] || './'
}
```
and then use only `target.filename` / `target.folder` instead of params.

**Comment by jotakar** [comment]
I always try to modify the minimal of code. This is just a line, and req.params takes its value in the same function (as a result of match the regexp), it is not readonly. This is the reason of wirting the code in this way. But yes, you're right a comment is good for everybody. 
Although if it's an obstacle for being commit I change the code, without any problem :)
Thanks for your attention.

**Comment by clemos** [comment]
Hi @jotakar 
I agree it's usually better to make minimal changes, but ITOH the current code is not exactly a collection of good practices, and some refactoring wouldn't hurt :)
If you are ok with [my suggestion](https://github.com/silexlabs/CloudExplorer2/pull/82/#issuecomment-431840900), I think it would be nice to implement it as part of your PR.
Otherwise, we'll do it and merge it eventually.
Thanks again :D

**Comment by jotakar** [comment]
Ok, it is a good reason. I change the code to @clemos  suggestion.
Thanks

**Comment by jotakar** [comment]
What about this request? is it fine to be commited?

**Comment by clemos** [comment]
Looks good to me :)

**Comment by lexoyo** [comment]
Hi @jotakar 
I am so sorry we forgot about this
This is what happens when a project has no official maintainer
@all I will merge this?

**Comment by JbIPS** [comment]
Just remove the console.log on line 251 and merge 😃

**Comment by lexoyo** [comment]
I think @jotakar may have moved on and forget about us
Sorry

**Comment by jotakar** [comment]
No, I'm here :)
But now I'm busy with other business. But I'm glad when I see this project is on and that changes where I collaborated are commited.

File: lib/router.js [hunk 1]
```diff
@@ -242,29 +242,35 @@ module.exports = class Router extends express.Router {
...
 
     this.post(/\/(.*)\/upload\/(.*)/, upload.array('content'), (req, res) => {
-      if (!req.body) Router.handleError(res, new Error('No body parsed supplied'));
+       if (!req.body) Router.handleError(res, new Error('No body parsed supplied'));
+      //empty path get a 400 error, so avoid it
+      const target = {
+          folder: req.params[0], 
+          path: req.params[1] || './'    
+      }
+  	  console.log("Desde el router ",req.session.unifile, req.params)
```
Comment (JbIPS) [inline lib/router.js]: Remove it.
```suggestion
  	 
```

**Comment by lexoyo** [comment]
I suppose we have all tested thoroughly so I have merged and will test now
Thx @jotakar for not giving up ;) And well done!

**Pull request merged**